fenix/app/src/androidTest/java/org/mozilla/fenix/ui/NimbusMessagingTest.kt

```diff
@@ -0,0 +1,233 @@
+        assertNotNull(message)
+    }
+
+    private suspend fun testMessageValidity() {
+        val messages = storage.getMessages()
+        val datas = FxNimbus.features.messaging.value().messages
+        assertTrue(datas.isNotEmpty())
+        assertEquals(messages.size, datas.size)
+
+        val helper =
+            nimbus.createMessageHelper(CustomAttributeProvider.getCustomAttributes(context))
+        messages.forEach { message ->
+            storage.isMessageEligible(message, helper)
+        }
+        assertTrue(storage.malFormedMap.isEmpty())
+    }
+
+    @Test
+    fun testValidNotificationMessage() = runTest {
+        hardcodedNimbus = HardcodedNimbusFeatures(
+            context,
+            "messaging" to JSONObject("""{
+              "message-under-experiment": "test-default-browser-notification",
+              "messages": {
+                "test-default-browser-notification": {
+                  "title": "preferences_set_as_default_browser",
+                  "text": "default_browser_experiment_card_text",
+                  "surface": "notification",
+                  "style": "NOTIFICATION",
+                  "action": "MAKE_DEFAULT_BROWSER",
+                  "trigger": [
+                    "ALWAYS"
+                  ]
+                }
+              }
+            }""".trimIndent(),
+            ),
+        )
+        FxNimbus.initialize { hardcodedNimbus }
+
+        val messages = storage.getMessages()
+        val message = storage.getNextMessage(MessageSurfaceId.NOTIFICATION, messages)
+        assertNotNull(message)
+        assertTrue(hardcodedNimbus.exposure.contains("messaging"))
+    }
+
+    @Test
+    fun testShowingNotificationMessage() {
+        hardcodedNimbus = HardcodedNimbusFeatures(
+            context,
+            "messaging" to JSONObject("""{
+              "notification-config": {
+                "refresh-interval": 0
+              },
+              "message-under-experiment": "test-default-browser-notification",
+              "messages": {
+                "test-default-browser-notification": {
+                  "title": "preferences_set_as_default_browser",
+                  "text": "default_browser_experiment_card_text",
+                  "surface": "notification",
+                  "style": "NOTIFICATION",
+                  "action": "MAKE_DEFAULT_BROWSER",
+                  "trigger": [
+                    "ALWAYS"
+                  ]
+                }
+              }
+            }""".trimIndent(),
+            ),
+        )
+        activityTestRule.finishActivity()
+        FxNimbus.initialize { hardcodedNimbus }
+        activityTestRule.launchActivity(null)
+
+        mDevice.openNotification()
+        notificationShade {
+            val data = FxNimbus.features.messaging.value().messages["test-default-browser-notification"]
+            verifySystemNotificationExists(data!!.title!!)
+            verifySystemNotificationExists(data.text)
```

Comment:
This detects when the notification is shown. 

It is working in Android API < O, but no later.

I don't know where the permission prompt is supposed to show in later versions. Does @rocketsroger or @rahulsainani know?

Reply:
What about for the homescreen message?

Reply:
Any device with Android 13 (Tieramisu) or above will need to user to opt-in (by showing the permission prompt first) before showing the notification.

Reply:
Ok, fixed it!

It turned out to be that the `MessageNotificationWorker` wasn't ensuring that the marketing channel existed. 

@rocketsroger Should this be called earlier (e.g. in `showNotificationPermissionPromptIfRequired()` for `SDK < TIRAMISU`)?